### ISSUE TITLE
Handle task errors

### ISSUE THREAD

**Issue by tcompa** [fractal-analytics-platform/fractal-server#106]
If we implement https://github.com/fractal-analytics-platform/fractal-server/issues/105, what should then happen (at the workflow level) if there is an error/exception in a given task? Should we stop all the following tasks.

**Comment by jluethi** [comment]
Yes, I would say if a task fails, the workflow should stop. Many later tasks rely on earlier tasks and would do weird things if they run, even though an intermediate failed.

If we think that would ever be different (or we'd like to include tasks that are optional), we could have this as a parameter. But the default should be that workflows stop. A user can then fix the cause of the error (e.g. change some parameter) and resume the workflow.

Also, if we end the workflow, we could avoid having this long error chain of "some other task before me failed" until we get to the actual error :)

**Comment by tcompa** [comment]
- [ ] Workflow job has a status, at least one of "submitted", "completed", "failed".
- [ ] (currently optional) Additional information would be useful ("submitted" -> "running task 3").
- [ ] If a task fails, workflow execution is stopped and status updated to "failed".
- [ ] User can query status of a workflow job, and if "failed" they can see 
    - [ ] which task failed
    - [ ] the error message or some logging information
